pennylane/tape/tapes/tape.py

```diff
@@ -941,9 +968,9 @@ def execute_device(self, params, device):
 
         if self._caching:
             circuit_hash = self.graph.hash
-            if circuit_hash in self._cache_execute:
+            if circuit_hash in self.get_cache():
```

Comment:
Are these changes needed for `copy()` to work?

Reply:
Yep! I will have a fully PR description comment ready soon 😆 Got bogged down fixing the copy issue